corehq/apps/cloudcare/static/cloudcare/js/formplayer/menus/controller.js

```diff
@@ -104,8 +106,8 @@ hqDefine("cloudcare/js/formplayer/menus/controller", function () {
 
         if (menuResponse.breadcrumbs) {
             menusUtils.showBreadcrumbs(menuResponse.breadcrumbs);
-            if (menuResponse.langs && menuResponse.langs.length > 1 && !appPreview && changeFormLanguage) {
-                menusUtils.showLanguageMenu(menuResponse.langs);
+            if (!appPreview && ((menuResponse.langs && menuResponse.langs.length > 1 && changeFormLanguage) || enablePrintOption)) {
+                menusUtils.showFormMenu(menuResponse.langs, changeFormLanguage);
```

Comment:
@stephherbers blast from the past here, but I just noticed that this logic is a little strange - the only place we check `enablePrintOption` is here, which means that even if it's false, we'll still display the print button if the app has multiple languages defined.  Do you recall what `!menuResponse.queryKey` is doing?  Are there situations where we're currently displaying the print option where we shouldn't be?

Reply:
queryKey is the "commands" like "m0-f1". I think the response from formplayer does not contain a `queryKey` when entering a form. so `!menuResponse.queryKey` signals that we're in a form. `langs` should only be in the menuResponse if we're in a form (based on [this](https://github.com/dimagi/commcare-hq/blob/e7391ddae1af1dbf118211ecb52c83fc508aa656/corehq/apps/cloudcare/static/cloudcare/js/formplayer/menus/collections.js#L70)) so with this logic, I think it'll only show the print when we're in a form (as expected)

Reply:
> `!menuResponse.queryKey` signals that we're in a form

I believe you, but there should be better ways to detect we're in a form, it's a different type of response.

Reply:
I was also thinking there should be a more reliable/intended way. I don't think there's a "type" in the response (if you meant like the `entities` and `query` types) It seems like how it's done [here](https://github.com/dimagi/commcare-hq/blob/734b20d54946b96c59b27907026efbbb22ca1b37/corehq/apps/cloudcare/static/cloudcare/js/formplayer/menus/collections.js#L118) is checking if a `response.tree` exists, where tree contains an array of questions/groups the form contains.

Reply:
Cool. Yeah, I was thinking of the `type` and thinking it was in form responses, but now that you mention it, looking for the `tree` attribute does sound familiar.